src/Metal/MTLIOCompression.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+#nullable enable
+
+#if !NET
+using NativeHandle = System.IntPtr;
+#endif
+
+namespace Metal {
+
+#if NET
+	[SupportedOSPlatform ("ios16.0")]
+	[SupportedOSPlatform ("maccatalyst16.0")]
+	[SupportedOSPlatform ("macos13.0")]
+	[SupportedOSPlatform ("tvos16.0")]
+#else
+	[Mac (13, 0), iOS (16, 0), TV (16, 0), MacCatalyst (16, 0)]
+#endif
+	public class MTLIOCompressionContext : DisposableObject {
+
+		[Preserve (Conditional = true)]
+		MTLIOCompressionContext (NativeHandle handle, bool owns) : base (handle, owns) { }
+
+		[DllImport (Constants.MetalLibrary)]
+		static extern unsafe void MTLIOCompressionContextAppendData (void* context, void* data, nuint size);
+
+		unsafe void AppendData (void* data, nuint size)
+			=> MTLIOCompressionContextAppendData ((void*) GetCheckedHandle (), data, size);
+
+		public void AppendData (byte [] data)
+			=> AppendData (new ReadOnlySpan<byte> (data, 0, data.Length));
+
+		public void AppendData (ReadOnlySpan<byte> data)
+		{
+			unsafe {
```

Comment:
Null check?

Reply:
ReadOnlySpan is a struct, not a ref type :)